packages/fx-core/src/plugins/resource/frontend/utils/progress-helper.ts

```diff
@@ -84,6 +89,19 @@ export class ProgressHelper {
     return this.deployProgress;
   }
 
+  static async startMigrateProgressHandler(
+    ctx: PluginContext
+  ): Promise<IProgressHandler | undefined> {
+    await this.migrateProgress?.end(true);
+
+    this.migrateProgress = ctx.ui?.createProgressBar(
+      Messages.MigrateProgressTitle,
+      Object.entries(MigrateSteps).length
+    );
+    await this.migrateProgress?.start(Messages.ProgressStart);
+    return this.migrateProgress;
+  }
+
   static async endAllHandlers(success: boolean): Promise<void> {
```

Comment:
Also end migrate progress here. This method will be invoked when error occurs.

Reply:
Got it.